app/jobs/delete_test_results_job.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+# Job deletes test results of a specified profile
+class DeleteTestResultsJob
+  include Sidekiq::Worker
+
+  def perform(profile_id)
+    TestResult.where(profile_id: profile_id).delete_all
```

Comment:
@bastilian I think this one needs to be `destroy_all` to trigger the `has_many :rule_results, dependent :delete_all`